sync_twindis/models/sync_project.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2021 Denis Mudarisov <https://github.com/trojikman>
+# License MIT (https://opensource.org/licenses/MIT).
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+from odoo.tools.translate import _
+
+
+class SyncProjectTwindis(models.Model):
+
+    _inherit = "sync.project"
+    eval_context = fields.Selection(
+        selection_add=[
+            ("twindis", "Twindis"),
+        ]
+    )
+
+    @api.model
+    def _eval_context_twindis(self, secrets, eval_context):
+        import base64
+        import csv
+        from io import BytesIO, TextIOWrapper
+
+        import requests
+
+        log = eval_context["log"]
+        params = eval_context["params"]
+        if not all(
+            [params.PRODUCTS_URL, params.PRICES_URL, secrets.LOGIN, secrets.PASSWORD]
+        ):
+            raise UserError(_("Credentials are not set"))
+
+        def read_csv_attachment(attachment):
+            """
+            :attachment: ir.attachment record with csv data
+
+            :return: tuple of csv fields and data
+            """
+            decoded_datas = base64.decodebytes(attachment.datas)
+            encoding = "utf-8"
+            f = TextIOWrapper(BytesIO(decoded_datas), encoding=encoding)
+            reader = csv.reader(f, delimiter=";")
+            fields = next(reader)
+            data = [row for row in reader]
+            return fields, data
```

Comment:
лучше передавать генератор, чтобы экономить процессор и память
```suggestion
            return fields, reader
```